### PR TITLE
add monthly income-expense report

### PR DESCRIPTION
**Pull request by kain88-de**
My monthly income and expenses vary a bit during the year. I usually generate this report manually to know which month has been expensive and check if the income looks reasonable for each month. This level of detail is a lot more helpful for me to see how I am doing in the current year.

I'm new to haskell. Are there any tests for the report that should be added to this PR as well?

### CODE REVIEW

**Comment by apauley** [comment]
> My monthly income and expenses vary a bit during the year. I usually generate this report manually to know which month has been expensive and check if the income looks reasonable for each month. This level of detail is a lot more helpful for me to see how I am doing in the current year.

Thanks for taking the time to submit this pull request @kain88-de, great stuff.
I can see that it would be useful to have monthly versions of the income statement (and possibly some of the other reports?)

> 
> I'm new to haskell. Are there any tests for the report that should be added to this PR as well?

I don't think we need more report tests yet, while we're still figuring out what reports people want.
We just need the existing tests to pass.

But I noticed the change didn't compile - luckily it looks as if we need a very small change to get it to compile.
I will comment on the line where the problem is, and then you can try again.

I noticed the commit was done via the Github web interface - can you compile it on your own machine?
The feedback cycle will be faster if you compile it yourself. 
Otherwise you can wait for the [Travis CI](https://travis-ci.org/github/apauley/hledger-flow/builds/717717814?utm_source=github_status&utm_medium=notification) and [CircleCI](https://circleci.com/gh/apauley/hledger-flow/249?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) results after each commit. But that will be slow.

File: src/Hledger/Flow/Reports.hs [hunk 1]
```diff
@@ -59,7 +59,7 @@ generateReports' opts ch = do
   ownerParams <- ownerParameters opts ch owners
   let ownerWithAggregateParams = (if length owners > 1 then [aggregateParams] else []) ++ ownerParams
   let sharedOptions = ["--pretty-tables", "--depth", "2"]
-  let ownerWithAggregateReports = List.concat $ fmap (reportActions opts ch [incomeStatement sharedOptions, balanceSheet sharedOptions]) ownerWithAggregateParams
+  let ownerWithAggregateReports = List.concat $ fmap (reportActions opts ch [incomeStatement incomeMonthlyStatement sharedOptions, balanceSheet sharedOptions]) ownerWithAggregateParams
```
Comment (apauley) [inline src/Hledger/Flow/Reports.hs:62]: The list `[incomeStatement sharedOptions, balanceSheet sharedOptions]` (baseline code) is a list of `ReportGenerator`s

In Haskell `incomeStatement sharedOptions` means we are supplying one argument (`sharedOptions`) to the function named `incomeStatement`, and in this case the result type is then `ReportGenerator`.

To use your new monthly function, I think you should add an element to the list like this:
`[incomeStatement sharedOptions, incomeMonthlyStatement sharedOptions, balanceSheet sharedOptions]`

**Comment by apauley** [comment]
Another reason to compile it yourself is so that you can run it first and look at the results. 

A lot of the time when I look at the output of what I did it triggers ideas to do it better, or I find out that I actually want to do something a bit differently.

File: src/Hledger/Flow/Reports.hs (hunk 1 above)
Comment (kain88-de) [reply src/Hledger/Flow/Reports.hs]: ah thanks, I overlooked that yesterday. This also explains where the sharedOptions come from. I was wondering where that variable comes from.

**Comment by kain88-de** [comment]
> Another reason to compile it yourself is so that you can run it first and look at the results.
> 
> A lot of the time when I look at the output of what I did it triggers ideas to do it better, or I find out that I actually want to do something a bit differently.

Normally I do that too. I'm just trying to do it extra hard and try to run all my builds in Docker containers or VMs.

**Comment by apauley** [comment]
Thanks @kain88-de, it is merged. I want to make a small change before this is released.

Instead of the current `reports/john/2020/income-expenses-monthly.txt` filename, I would like to make a `monthly` directory where this and future monthly reports can live.

The above example would then become `reports/john/2020/monthly/income-expenses.txt`.

If you want to do it please let me know, otherwise I will look at it when I get a chance.

**Comment by kain88-de** [comment]
I'm not super into that idea. I thought it could be possible to place the reports into a config file. I do not see why this feature should require to recompile the program to add a new report. The program could have a build-in standard config for the current reports.

For the config, I would tend to simple key:value design with `filename:hledger options`. But it means it cannot say in which folder the file should be saved. We could also use a more verbose config that stores a folder to save the file. Or if the report should be calculated for all accounts or individual ones.

**Comment by apauley** [comment]
@kain88-de All of your suggestions sound good to me. 

Adding some config is something I've thought about doing in the past (see #18), but with the amount of free time I currently have to spend on this project I haven't been able to do it.

**Comment by apauley** [comment]
I've changed the location of the monthly income statement to be in a `monthly` directory:

https://github.com/apauley/hledger-flow/commit/56554ddc388d87b73d36180716ba7db036c018af

**Comment by kain88-de** [comment]
I can give it a try. But I’m not entirely sure how much time I’ll have either.

**Pull request merged**